core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/OneOrMore.java

```diff
@@ -101,6 +102,7 @@ public boolean addAll(final Collection<? extends T> items) {
         return result;
     }
 
+    @SuppressFBWarnings(value = "BC_BAD_CAST_TO_ABSTRACT_COLLECTION", justification = "Cast to List is required to use a List specific remove method")
     public void removeAnyItem() {
         if (null == collection) {
             singleItem = null;
```

Comment:
Maybe it could get put on two lines instead? and also check that we can cast to List?

Reply:
The `@SuppressFBWarnings` on two lines? Checking if we can cast to List won't affect the warning.

Reply:
Yeah if it can maybe the line would be a bit less long

Reply:
I can't work out why this (casting to list if deduplicate is false) to use the remove(int i) method is being used in this case over the iterator method used for deduplicate. What am I missing here?

Reply:
@t11947 When deduplicate is false it means the `Collection` is always a list. I think it could probably just use the iterator. However there is a difference in the behaviour at least. For the list it removes the last item, but when the iterator is used I think it's removing the first item. Given this method is called `removeAnyItem` how this works is pretty odd.

I'm not sure what would happen if we started removing from the start of lists instead of the end, but it could cause strange things to happen, so probably best to keep the current suboptimal situation!